src/pluginHooks.js

```diff
@@ -1907,6 +1907,17 @@ const REGISTERED_HOOKS = [
    * @param {boolean} successfullyExpanded `true`, if the action affected any non-collapsible column, `false` otherwise.
    */
   'afterColumnExpand',
+
+  /**
+   * Fired by {@link AutoColumnSize} plugin within {@link SampleGenerator} utility.
+   *
+   * @event Hooks#modifyAutoColumnSizeSeed
+   * @since 8.4.0
+   * @param {number|string|undefined} seed Seed ID, unique name to categorize samples.
```

Comment:
Not always a name, this may be a number that is used to increase the id? I think that internally it worked like that.

BUT... As it's not yet public, it's not too late to change this and the checkbox behavior to use `string` only.

Reply:
`name` seemed to me as the best fitting description because the internal map allows using number and string at the same time.

Do you suggest rewriting that fragment of code where we are using `length` to define seed?

Reply:
I'm talking about SamplesGenerator where it will try to add seeds:
https://github.com/handsontable/handsontable/blob/a58d4e235c90bb896d7d7488b7a00c197cffa1a4/src/utils/samplesGenerator.js#L156-L158
It might have been considered previously as a kind of "numeric seed" that have a "modifier", not a unique string id. You have renamed it from `length` to `seed` but the logic has not changed.

Two problems here:
1. This complicates the usage. Sometimes it's a number (checkbox hook will return `length`), sometimes a string. This may confuse the developer which one should she use. If she may return string when number was given.
2. There might be a use case scenario when I want to return `my-custom-seed` from the hook, but internally SampleGenerator will add `length + "my-custom-seed"` and I will have multiple seeds, when I wanted a shared one. ie. `1my-custom-seed`, `200my-custom-seed` etc. This may also confuse the developer, have to known inner logic of SamplesGenerator to predict the outcome.

Reply:
I'm putting it for a discussion. Maybe if `bundleCountSeed` was defined it should overwrite anything else:

```diff
+      if (hasCustomBundleSeed) {
+        seed = bundleCountSeed;
+      }
-      if (isObject(value)) {
+      else if (isObject(value)) {
        seed = Object.keys(value).length;

      } else if (Array.isArray(value)) {
        seed = value.length;

      } else {
        seed = stringify(value).length;
      }

-      if (hasCustomBundleSeed) {
-        seed += bundleCountSeed;
-      }
```

Reply:
Yeah, it seems reasonable to move all `seed`-related things to string. Please check the newest changes. https://github.com/handsontable/handsontable/pull/7633/commits/4d00db174e3b6f7b056b4165713512905418e805#diff-c1ecea5accb17187b7e2e83ff8f2a3e7e9ed5e1329767c21b0e77d6bad49885e